Vates/VatesAPI/src/vtkMDHexFactory.cpp

```diff
@@ -117,16 +117,18 @@ void vtkMDHexFactory::doCreate(
       if (!isSpecial(signal_normalized) &&
           m_thresholdRange->inRange(signal_normalized)) {
         // Cache the signal and using of it
-        signalArray[i] = float(signal_normalized);
+        signalCache[i] = float(signal_normalized);
         useBox[i] = true;
 
         // Get the coordinates.
         size_t numVertexes = 0;
-        coord_t *coords;
+        coord_t *coords; // raw pointer as comment below suggests it is
+                         // speed-critical
```

Comment:
Are you sure? The overhead of `std::unique_ptr` is very small. And there is a potentially long loop over `numVertexes` below where the smart pointer is not needed directly, which should completely hide the cost of the unique_ptr.